Packages/com.unity.inputsystem/InputSystem/InputSystem.cs

```diff
@@ -3519,7 +3520,6 @@ private static void Destroy()
             //       state repeatedly during tests but we want to not create InputSystemObject
             //       over and over.
 
-            InputActionState.ResetGlobals();
```

Comment:
@Rene-Damm With these changes, I think this is not needed. Do you agree?

Reply:
Seems correct.

However, seems that the ordering has changed now. Seems we now *first* destroy `InputManager` and then reset the saved state. That means that `InputActionState` et al will now do its work against an `InputManager` that has technically been destroyed. Seems problematic.

Reply:
hmmm.... will take a look

Reply:
Hopefully this works correctly now? @Rene-Damm 
This makes the whole interface/abstraction thing kind of redundant, but should probably be revisited together with an attempt to make life-time management consistent, do you agree?